added STmag to fnu conversion example

johnnygreco opened the pull request:
I added an example showing how to convert magnitudes to flux densities using the appropriate equivalency. While it was possible to figure out how to do this based on the existing documentation, I think an explicit example is still quite useful. 

Also, I removed an extra comma in the coordinates documentation, which was from an example I wrote earlier this year.

pllim commented on docs/units/logarithmic_units.rst:
```diff
@@ -137,6 +137,29 @@ appropriate equivalency::
+To convert ``V`` to flux density per unit frequency (on which AB magnitudes
+are based), we again need the appropriate equivalency::
+
+    >>> fnu = V.to(u.erg/u.s/u.cm**2/u.Hz, u.spectral_density(5500*u.AA))
```
Why not define `lam = 5500 * u.AA` before this line and then use it here as well?
```suggestion
    >>> lam = 5500 * u.AA
    >>> fnu = V.to(u.erg/u.s/u.cm**2/u.Hz, u.spectral_density(lam))
```

pllim commented on docs/units/logarithmic_units.rst:
```diff
@@ -137,6 +137,29 @@ appropriate equivalency::
+are based), we again need the appropriate equivalency::
+
+    >>> fnu = V.to(u.erg/u.s/u.cm**2/u.Hz, u.spectral_density(5500*u.AA))
+    >>> fnu # doctest: +FLOAT_CMP
```
```suggestion
    >>> fnu  # doctest: +FLOAT_CMP
```

pllim commented on docs/units/logarithmic_units.rst:
```diff
@@ -137,6 +137,29 @@ appropriate equivalency::
+
+We could have also used frequency for the equivalency::
+
+    >>> lam = 5500*u.AA
```
Moved the definition to above.
```suggestion
```

pllim commented on docs/units/logarithmic_units.rst:
```diff
@@ -137,6 +137,29 @@ appropriate equivalency::
+We could have also used frequency for the equivalency::
+
+    >>> lam = 5500*u.AA
+    >>> nu = c.c.to('AA/s') / lam
```
Calculating `nu` like this is confusing to me. If you already know `lam`, there is no need to calculate `nu` and then redo the same conversion. So it makes more sense if you hardcode the value of `nu` here and use it directly. To make it clear that this only makes sense if user already have `nu` anyway.

johnnygreco replied:
Hard coding sounds good to me! 

```suggestion
    >>> nu = 5.45077196e+14 * u.Hz
```

pllim commented on docs/units/logarithmic_units.rst:
```diff
@@ -137,6 +137,29 @@ appropriate equivalency::
+    >>> lam = 5500*u.AA
+    >>> nu = c.c.to('AA/s') / lam
+    >>> fnu = V.to(u.erg/u.s/u.cm**2/u.Hz, u.spectral_density(nu))
+    >>> fnu # doctest: +FLOAT_CMP
```
```suggestion
    >>> fnu  # doctest: +FLOAT_CMP
```

johnnygreco commented on docs/units/logarithmic_units.rst:
```diff
@@ -137,6 +137,29 @@ appropriate equivalency::
+    >>> flam # doctest: +FLOAT_CMP
+    <Quantity [5.75439937e-16, 1.29473986e-17, 3.59649961e-18] erg / (Angstrom cm2 s)>
+
+To convert ``V`` to flux density per unit frequency (on which AB magnitudes
```
I wanted to highlight that if one had AB mags, they wouldn't need an equivalency to convert to f_nu, but I agree it is confusing here. Let's delete it.
```suggestion
To convert ``V`` to flux density per unit frequency,
```

bsipocz commented:
Commits here needs to be squashed together before merging. 

@johnnygreco - let me know if you need any help with that, happy to walk you though on the astropy slack, or do it for you.

johnnygreco commented:
I just squashed the commits; first time I've done this, so please let me know if I messed it up somehow! I also added a note from a comment on Twitter: https://twitter.com/tim_carleton/status/1142827983478120449 

Thanks!

kelle commented on docs/units/logarithmic_units.rst:
```diff
@@ -137,6 +137,35 @@ appropriate equivalency::
 
+This is particularly useful for converting magnitude into flux density.
+``V`` is currently in ST magnitudes, which is based on flux densities per
+unit wavelength::
```
To covert ``V`` into a flux density per wavelength, ``f_lam``::

johnnygreco replied:
```suggestion
unit wavelength (:math:`f_\lambda`). Therefore, we can directly convert ``V`` into flux density per unit wavelength using the :meth:`~astropy.units.quantity.Quantity.to` method::
```

kelle commented on docs/units/logarithmic_units.rst:
```diff
@@ -137,6 +137,35 @@ appropriate equivalency::
+    <Quantity [5.75439937e-16, 1.29473986e-17, 3.59649961e-18] erg / (Angstrom cm2 s)>
+
+To convert ``V`` to flux density per unit frequency, we again need the 
+appropriate equivalency::
```
and the central wavelength of the magnitude band, 5500 Angstroms, in this case::

johnnygreco replied:
```suggestion
appropriate :ref:`equivalency <unit_equivalencies>`, which in this case, is the central wavelength of the magnitude band, 5500 Angstroms::
```

kelle commented on docs/units/logarithmic_units.rst:
```diff
@@ -137,6 +137,35 @@ appropriate equivalency::
+    >>> fnu  # doctest: +FLOAT_CMP
+    <Quantity [5.80636959e-27, 1.30643316e-28, 3.62898099e-29] erg / (cm2 Hz s)>
+
+We could have also used frequency for the equivalency::
```
the central frequency instead

johnnygreco replied:
```suggestion
We could have used the central frequency instead::
```

kelle commented on docs/units/logarithmic_units.rst:
```diff
@@ -137,6 +137,35 @@ appropriate equivalency::
+.. Note::
+
+    When converting magnitudes to flux densities, order of operations matters.
+    For example, ``21 * u.ABmag.to(u.erg/u.s/u.cm**2/u.Hz)`` is not the same as 
```
In order to convert an AB mag of 21 to <something else>, the value and the unit need to be established before the conversion. For example, 21 * blah blah will give you 21 times an AB mag of 1 while (21*u.ABmag) in <blah units> will give you ABmag of 21 in <blah units>.

johnnygreco commented:
Great suggestions @kelle! I've incorporated them and squashed the commits! Thanks!

kelle commented:
Looks great to me! Thanks so much for the contribution. And thanks to @eilatg for the original question! :)

lglattly commented on docs/units/logarithmic_units.rst:
```diff
@@ -137,6 +137,40 @@ appropriate equivalency::
+    >>> flam  # doctest: +FLOAT_CMP
+    <Quantity [5.75439937e-16, 1.29473986e-17, 3.59649961e-18] erg / (Angstrom cm2 s)>
+
+To convert ``V`` to flux density per unit frequency  (:math:`f_\nu`), we again need 
```
```suggestion
To convert ``V`` to flux density per unit frequency (:math:`f_\nu`), we again need 
```

lglattly commented on docs/units/logarithmic_units.rst:
```diff
@@ -137,6 +137,40 @@ appropriate equivalency::
+    <Quantity [5.75439937e-16, 1.29473986e-17, 3.59649961e-18] erg / (Angstrom cm2 s)>
+
+To convert ``V`` to flux density per unit frequency  (:math:`f_\nu`), we again need 
+the appropriate :ref:`equivalency <unit_equivalencies>`, which in this case, is the 
```
```suggestion
the appropriate :ref:`equivalency <unit_equivalencies>`, which in this case is the 
```

lglattly commented on docs/units/logarithmic_units.rst:
```diff
@@ -137,6 +137,40 @@ appropriate equivalency::
+
+.. Note::
+
+    When converting magnitudes to flux densities, order of operations matters;
```
```suggestion
    When converting magnitudes to flux densities, the order of operations matters;
```

lglattly commented:
I made some small editing suggestions -- looks good to me otherwise!

johnnygreco commented:
Thanks @lglattly!

bsipocz commented:
Thank you @johnnygreco!
